src/components/pages/profile/ChangePassword.tsx

```diff
@@ -16,15 +19,41 @@ export const ChangePasswordModal = ({
   isOpen,
   onClose,
 }: ChangePasswordModalProps) => {
+  const handleClose = () => {
+    resetForm();
+    onClose();
+  };
+
+  const { mutate, isPending } = useChangePassword(() => handleClose());
+
+  const { handleSubmit, register, isValid, resetForm } = useFormikWrapper({
+    validateOnMount: true,
+    initialValues: {
+      old_password: "",
+      new_password: "",
+      confirm_password: "",
+    },
+    validationSchema: changePasswordSchema,
+    onSubmit(values) {
+      mutate({
+        old_password: values.old_password,
+        new_password: values.new_password,
+      });
+    },
+  });
+
   return (
     <Dialog
       open={isOpen}
-      onClose={onClose}
+      onClose={handleClose}
       as="div"
       className="relative z-10 focus:outline-none"
     >
       <div className="fixed inset-0 z-10 w-screen overflow-scroll scrollbar-hide duration-300 ease-out transition-opacity data-[closed]:opacity-0 bg-grey-dark-4/70">
-        <div className="flex flex-col min-h-full items-center p-3 justify-end md:justify-center">
+        <form
+          onSubmit={handleSubmit}
+          className="flex flex-col min-h-full items-center p-3 justify-end md:justify-center"
+        >
```

Comment:
@AdejokeOgunyinka Instead of this, the `<DialogPanel />` has a prop called `as` that takes a `String | Component`, so you can pass `<DialogPanel as="form" />`